File: StylingEngine/StylingEngine/ViewController.m
//
//  ViewController.m
//  StylingEngine
//
//  Created by Ruud Puts on 17/07/2018.
//  Copyright © 2018 Ruud Puts. All rights reserved.
//

#import "ViewController.h"
#import "StylingEngine-Swift.h"

@interface ViewController ()

@property (nonatomic, weak) IBOutlet UIView *topView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];


    UILabel *label = [UILabel new];
    [self.view addSubview:label];

    [self.topView styleAs:JMBViewStylePromotionView];
    [self.view styleAs:JMBViewStyleBackgroundView];
    [label styleAs:JMBViewStylePromotionLabel];
    /**
     Or in Swift:
     self.topView.style(as: .promotionView)
     self.view.style(as: .backgroundView)
     label.style(as: .backgroundView)
     **/

    label.text = @"I'm a label, look at me shine 🌟";
    [label sizeToFit];
    label.center = self.view.center;
}

@end

File: ModuleArchitecture/Modules/FeatureModule/FeatureModule/Source/ViewController.swift
//
//  ViewController.swift
//  FeatureModule
//
//  Created by Ruud Puts on 19/01/2019.
//  Copyright © 2019 MobileSorcery. All rights reserved.
//

import UIKit
import UIModule
import NetworkModule

class ViewController: UIViewController {

    @IBOutlet weak var imageView: UIImageView!

    override func viewDidLoad() {
        super.viewDidLoad()

        UIModule.test()
        NetworkModule.test()

        view.style(as: .testView)

        Request.get(url: "https://api.ipify.org?format=json") {
            NSLog("Response: \($0 ?? "-")")
        }
        
        imageView.image = UIImage(named: "burger", in: FeatureModule.bundle, compatibleWith: nil)
    }

    @IBAction func doTheThing(_ sender: Any) {
    }
}

extension ViewStyling where ViewType == UIView {
    static var testView = ViewStyling {
        $0.backgroundColor = UIColor.purple
    }
}
